server/vulnerabilities/msrc/analyzer.go

```diff
@@ -49,62 +48,39 @@ func Analyze(
 	toInsert := make(map[string]fleet.OSVulnerability)
 	toDelete := make(map[string]fleet.OSVulnerability)
 
-	var offset int
-	for {
-		hIDs, err := ds.HostIDsByOSID(ctx, os.ID, offset, hostsBatchSize)
-		if err != nil {
-			return nil, err
-		}
+	// Run vulnerability detection for all hosts in this batch (hIDs)
+	// and store the results in 'found'.
+	var found []fleet.OSVulnerability
```

Comment:
Do we test this `Analyze` function? If not, can we add a test?

Reply:
That's a good idea, we have similar tests with NVD that use static test data.  But I may pull that into a new task.